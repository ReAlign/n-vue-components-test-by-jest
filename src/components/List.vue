<template>
  <div>
    <ul>
      <li v-for="item in items">
        {{ item }}
      </li>
    </ul>
    {{ count }}
    <button @click="increment">Increment</button>
  </div>
</template>

<script>
export default {
  name: 'list',
  props: [
    'items'
  ],
  data() {
    return {
      count: 1
    }
  },

  methods: {
    increment () {
      this.count++;
    }
  }
}
</script>
