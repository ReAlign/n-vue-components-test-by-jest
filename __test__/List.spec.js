import { shallow, mount } from 'vue-test-utils'
import { createRenderer } from 'vue-server-renderer'

import List from '@/components/List.vue'

describe('List.vue', () => {
  it('renders li for each item in props.items', () => {
    const items = ['', '', '']
    const wrapper = shallow(List, {
      propsData: { items }
    })
    expect(wrapper.findAll('li')).toHaveLength(items.length)
  })

  it('matches snapshot', () => {
    const items = ['item 1', 'item 2', 'item 3']
    const renderer = createRenderer()
    const wrapper = shallow(List, {
      propsData: { items }
    })
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })

  it('button should increment the count', () => {
    const wrapper = mount(List)
    expect(wrapper.vm.count).toBe(0)
    const button = wrapper.find('button')
    button.trigger('click')
    expect(wrapper.vm.count).toBe(1)
  })
})
