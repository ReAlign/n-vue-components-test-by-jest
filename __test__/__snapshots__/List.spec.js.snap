// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List.vue matches snapshot 1`] = `
<div>
    <ul>
        <li>
            item 1
        </li>
        <li>
            item 2
        </li>
        <li>
            item 3
        </li>
    </ul>
    0
    <button>Increment</button></div>
`;
